views/index.erb

```diff
@@ -9,3 +12,5 @@
 	      </a>
       </div>
 <% end %>
+</div>
+<%= alert(@error_msg) if !@error_msg.empty? %> 
```

Comment:
This logic should be moved into the controller. I would create . `@clusters` instance variable in the `get '/clusters' do` section that does something like this:

```ruby
@clusters = CLUSTERS.map { | cluster| MoabShowqClient.new(cluster).setup }
```

Now you have an array of either `MoabShowqClient` or `MoabShowqClientNotAvailable` instances.

1. For each cluster you can render the div.
2. For each cluster you can render the error message.

Also, I noticed that here you are making a human readable form of the `MoabShowqClient` by doing this:

```ruby
@error_msg.concat("#{cluster.metadata.title|| cluster.id.titleize} Cluster: #{(MoabShowqClient.new(cluster).setup).error_message}</br>")
```

So the error message says something like: "Owens Cluster: #{error_message}".

This logic should be in the model too. If `MoabShowqClient#error_message` contains just the actual error message produced, then maybe `MoabShowqClient#friendly_error_message` or `MoabShowqClient#error_message_humanized` would return the string with the prefix, if the error_ message exists. That way you could display an error message for each cluster as appropriate.

Reply:
Oh yeah, i realize after seeing your screenshot you are collecting all of the error messages together in one alert. If it makes the code easier, you can just have 3 separate alerts.